server/routes/auth/auth.js

```diff
@@ -0,0 +1,63 @@
+const express = require('express');
+const passport = require('passport');
+const bcrypt = require('bcryptjs');
+const router = express.Router();
+
+const User = require('../../database/models/User');
+
+const saltRounds = 12;
+
+router.route('/register')
+.post((req, res) => {
+
+  console.log('/register post request, ', req.body);
+  bcrypt.genSalt(saltRounds, (error, salt) => {
+    if (error) { console.log('genSalt error ', error); } //return 500
+
+    bcrypt.hash(req.body.password, salt, (error, hash) => {
+      if (error) { console.log('hash error ', error); } //return 500
+
+      return new User({ // If no errors were encountered salting and hashing the password then create a new model.
+        
+        role_id: 3, // Basic User
+        active: true,
+        theme_id: 1,
+        username: req.body.username,
+        profileImageUrl: user.profileImageUrl,
```

Comment:
user is uploading a profileImageUrl so this should come from req.body.profileImageUrl.  We haven't covered how to upload images to AWS S3 yet but the incoming information should be similar.